Tests/NSUserDefaults+MMEConfigurationTests.m

```diff
@@ -87,6 +87,77 @@ - (void)testConfigEventTagDefault {
     XCTAssert([NSUserDefaults.mme_configuration.mme_eventTag isEqual:@"tag"]);
 }
 
+- (void)testCustomProfileOverMaxValues {
+    NSBundle.mme_mainBundle = [MMEBundleInfoFake bundleWithFakeInfo:@{
+        MMEEventsProfile: MMECustomProfile,
+        MMEStartupDelay : @9001,
+        MMECustomGeofenceRadius: @9001
+    }];
+    [NSUserDefaults.mme_configuration mme_registerDefaults];
+
+    XCTAssert(NSUserDefaults.mme_configuration.mme_backgroundGeofence == 300);
+    XCTAssert(NSUserDefaults.mme_configuration.mme_startupDelay == 1);
+}
+
+- (void)testCustomProfileUnderMinValues {
+    NSBundle.mme_mainBundle = [MMEBundleInfoFake bundleWithFakeInfo:@{
+        MMEEventsProfile: MMECustomProfile,
+        MMEStartupDelay : @-42,
+        MMECustomGeofenceRadius: @10
+    }];
+    [NSUserDefaults.mme_configuration mme_registerDefaults];
+
+    XCTAssert(NSUserDefaults.mme_configuration.mme_backgroundGeofence == 300);
+    XCTAssert(NSUserDefaults.mme_configuration.mme_startupDelay == 1);
+}
+
+- (void)testCustomProfile {
+    NSBundle.mme_mainBundle = [MMEBundleInfoFake bundleWithFakeInfo:@{
+        MMEEventsProfile: MMECustomProfile,
+        MMEStartupDelay : @2,
+        MMECustomGeofenceRadius: @420
+    }];
+    [NSUserDefaults.mme_configuration mme_registerDefaults];
+
+    XCTAssert(NSUserDefaults.mme_configuration.mme_backgroundGeofence == 420);
+    XCTAssert(NSUserDefaults.mme_configuration.mme_startupDelay == 2);
+}
+
+- (void)testCustomProfileInvalidValues {
+    NSBundle.mme_mainBundle = [MMEBundleInfoFake bundleWithFakeInfo:@{
+        MMEEventsProfile: MMECustomProfile,
+        MMEStartupDelay : @"unicorn",
+        MMECustomGeofenceRadius: @"fence"
+    }];
+    [NSUserDefaults.mme_configuration mme_registerDefaults];
+
+    XCTAssert(NSUserDefaults.mme_configuration.mme_backgroundGeofence == 300);
+    XCTAssert(NSUserDefaults.mme_configuration.mme_startupDelay == 1);
+}
+
+- (void)testDefaults {
+    NSBundle.mme_mainBundle = [MMEBundleInfoFake bundleWithFakeInfo:@{
+        MMEAccountType: @1,
+        @"MMEMapboxUserAgentBase" : @"com.mapbox.test",
+        @"MMEMapboxHostSDKVersion": @"1.0.0"
+    }];
+    [NSUserDefaults.mme_configuration mme_registerDefaults];
+
+    XCTAssertFalse(NSUserDefaults.mme_configuration.mme_isCollectionEnabled);
+    XCTAssert([NSUserDefaults.mme_configuration.mme_legacyUserAgentBase isEqualToString:@"com.mapbox.test"]);
+    XCTAssert([NSUserDefaults.mme_configuration.mme_legacyHostSDKVersion isEqualToString:@"1.0.0"]);
+}
+
+//TODO: Persistent object deletion isn't working as expected
```

Comment:
We should resolve this before merging these changes

Reply:
@rclee looking at testDeletePersistentObject we probably shoudn’t be mixing that test with MMECollectionDisabled which has some special cases

I suggest re-writing the test to use mme_setObject:forPersistentKey: (which you’ll need to add to the private header, maybe along with it’s volatile counterpart

Reply:
Done, thanks!